vault/vault.go

```diff
@@ -0,0 +1,183 @@
+package vault
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/hashicorp/vault/api"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	kubernetesJwtTokenPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	kubernetesAuthUrl      = "auth/kubernetes/login"
+)
+
+var log logr.Logger
+
+func fileExists(name string) (bool, error) {
+	_, err := os.Stat(name)
+	if err == nil {
+		return true, nil
+	}
+	if errors.Is(err, os.ErrNotExist) {
+		return false, nil
+	}
+	return false, err
+}
+
+func vaultClient() (*api.Client, error) {
+	var client *api.Client
+	var err error
+	var vaultSkipVerify bool = false
+
+	vaultUrl := os.Getenv("VAULT_ADDR")
+	if os.Getenv("VAULT_SKIP_VERIFY") != "" && os.Getenv("VAULT_SKIP_VERIFY") == "true" {
+		vaultSkipVerify = true
+	}
+	if vaultUrl == "" {
+		return client, fmt.Errorf("VAULT_ADDR is not set")
+	}
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: vaultSkipVerify},
+	}
+
+	httpClient := &http.Client{Transport: tr}
+
+	// create a vault client
+	client, err = api.NewClient(&api.Config{Address: vaultUrl, HttpClient: httpClient})
```

Comment:
Here you could just do `return api.NewClient()` and do away with the `client` variable altogether